src/main/java/com/github/fge/fs/dropbox/driver/DropboxWatchService.java

```diff
@@ -192,24 +195,26 @@ public WatchKey poll(final long timeout, final TimeUnit unit) throws Interrupted
     */
    @Override
    public WatchKey take() throws InterruptedException {
+	   
       if (_httpClient == null) {
-         return null;
+    	  throw new ClosedWatchServiceException();
       }
-
+      
       _hasChanges = new AtomicBoolean(false);
       _continuePolling = true;
 
       while (_hasChanges.get() == false && _continuePolling) {
-         try {
-
+         try {        	 
+             if (_httpClient == null) {            	 
+            	 throw new ClosedWatchServiceException();
+             }      
             final String cursor = getLatestCursor("");
 
-            //  final ListFolderLongpollResult listFolderLongpollResult = DropboxClient.getDefault().files().listFolderLongpoll(cursor);
```

Comment:
@rtdog  I will keep those commented lines as it was on purpose.
If Dropbox fulfills [this request in their SDK](https://github.com/FJBDev/java7-fs-dropbox/blob/d5ce6f59e7f4f1e31e0dd8694778995bed3d9f98/src/main/java/com/github/fge/fs/dropbox/driver/DropboxWatchService.java#L135), then I can uncomment this code